stat.go

```diff
@@ -125,6 +125,32 @@ func ChiSquare(obs, exp []float64) float64 {
 	return result
 }
 
+// CircularMean returns the circular mean of the dataset.
```

Comment:
I added the formula, tried to rely on the formatting from the other formulas in `stats.go`.